### PR TITLE
resovle the issue "Reading variables without loading entire Netcdf File."

### PR DESCRIPTION
**Pull request by qlongyinqw**
By using the feature in the commits,we are able to read netcdf file on HDFS like on local filesystem(only load necessary bytes).As with the demo code, we can process one big netcdf file in spark cluster(different spark node to process different variable or time data in the netcdf file). Finally, forgive me my poor English.

```
def main(args:Array[String]): Unit ={

  val scConf = new SparkConf().setAppName("hdfs nc").setMaster("local[3]")
  val sc = new SparkContext(scConf)

  val fsUri = "hdfs://master:9000"
  val hdfsFile = "/netcdf/HS_H08_20151130_0000_B01_FLDK_R10_S0110.nc"

  val nc = NetCDFUtils.loadDFSNetCDFDataSet(fsUri,hdfsFile,10000)

  val variables = nc.getVariables.map(_.getFullName)
  val varRDD = sc.parallelize(variables,3)

  val varSize = varRDD.map(v => {
    val nc = NetCDFUtils.loadDFSNetCDFDataSet(fsUri,hdfsFile,10000000)
    val variable = nc.findVariable(v)

    //do some process
    (v,variable.read().getSize)
  }).collect()

  varSize.foreach(vs => {
    println("variable name:"+vs._1+"\tsize:"+vs._2)
  })
}
```

### CODE REVIEW

**Comment by lewismc** [comment]
Do you have a description of the problem here please @qlongyinqw ?
Is it possible for you to squash your commits into one clean one? Please see [here](https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/)

File: src/main/java/org/dia/HDFSRandomAccessFile.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
+package orq.dia;
```
Comment (chrismattmann) [inline src/main/java/org/dia/HDFSRandomAccessFile.java:18]: needs ALv2 license on this.

File: src/main/scala/org/dia/Utils/NetCDFUtils.scala [hunk 2]
```diff
@@ -122,6 +124,25 @@ object NetCDFUtils {
     }
   }
 
+  /**
+    * Loads a NetCDF Dataset from HDFS
+    * @param dfsUri   HDFS URI(eg. hdfs://master:9000/)
+    * @param location file path on hdfs
+    * @param bufferSize the size of buffer to be used
+    */
+  def loadDFSNetCDFDataSet(dfsUri:String,location:String,bufferSize:Int):NetcdfDataset = {
```
Comment (chrismattmann) [inline src/main/scala/org/dia/Utils/NetCDFUtils.scala:133]: very cool!

**Comment by chrismattmann** [comment]
agree with @lewismc interested in knowing the specific use case this solves @qlongyinqw

File: build.sbt [hunk 3]
```diff
@@ -13,7 +13,8 @@ mainClass in Compile := Some("org.dia.algorithms.mcc.MainNetcdfDFSMCC")
 
 resolvers ++= Seq(
   Resolver.mavenLocal,
-  Resolver.file("Local", file(Path.userHome.absolutePath + "/.ivy2/local"))(Resolver.ivyStylePatterns)
+  Resolver.file("Local", file(Path.userHome.absolutePath + "/.ivy2/local"))(Resolver.ivyStylePatterns),
+  "maven Repository" at "http://artifacts.unidata.ucar.edu/content/repositories/unidata-releases/"
```
Comment (qlongyinqw) [inline build.sbt]: add ucar repository

**Comment by qlongyinqw** [comment]
I have modified the description of the pull request and squash my commits into clean one @lewismc 
I have added the ALv2 license on it @chrismattmann

**Comment by chrismattmann** [comment]
thanks @qlongyinqw committing now!

**Pull request merged**